contracts/allocators/FuseAllocator.sol

```diff
@@ -0,0 +1,539 @@
+
+    function getCash() external view returns (uint256);
+
+    function accrueInterest() external returns (uint256);
+
+    function seize(
+        address liquidator,
+        address borrower,
+        uint256 seizeTokens
+    ) external returns (uint256);
+
+    /*** Admin Functions ***/
+
+    function _setPendingAdmin(address payable newPendingAdmin) external returns (uint256);
+
+    function _acceptAdmin() external returns (uint256);
+
+    function _setComptroller(ComptrollerInterface newComptroller) external returns (uint256);
+
+    function _setReserveFactor(uint256 newReserveFactorMantissa) external returns (uint256);
+
+    function _reduceReserves(uint256 reduceAmount) external returns (uint256);
+
+    function _setInterestRateModel(InterestRateModel newInterestRateModel) external returns (uint256);
+}
+
+interface CErc20Interface is CTokenInterface {
+    /*** User Interface ***/
+
+    function mint(uint256 mintAmount) external returns (uint256);
+
+    function redeem(uint256 redeemTokens) external returns (uint256);
+
+    function redeemUnderlying(uint256 redeemAmount) external returns (uint256);
+
+    function borrow(uint256 borrowAmount) external returns (uint256);
+
+    function repayBorrow(uint256 repayAmount) external returns (uint256);
+
+    function repayBorrowBehalf(address borrower, uint256 repayAmount) external returns (uint256);
+
+    function liquidateBorrow(
+        address borrower,
+        uint256 repayAmount,
+        CTokenInterface cTokenCollateral
+    ) external returns (uint256);
+
+    function sweepToken(EIP20NonStandardInterface token) external;
+
+    /*** Admin Functions ***/
+
+    function _addReserves(uint256 addAmount) external returns (uint256);
+}
+
+contract FuseAllocator is OlympusAccessControlled {
+    /* ======== DEPENDENCIES ======== */
+
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ======== STATE VARIABLES ======== */
+    ITreasury public treasury; // Olympus Treasury
+
+    /* ======== CONSTRUCTOR ======== */
+
+    constructor(address _treasury, IOlympusAuthority _authority) OlympusAccessControlled(_authority) {
+        setTreasury(_treasury);
+    }
+
+    /* ======== POLICY FUNCTIONS ======== */
+
+    /**
+     * @notice Deposit the tokens in exchange for the corresponding fTokens,
+     * then enter the designated pool with those fTokens.
+     * @dev The amount of fTokens received is the number of tokens divided by the exchange rate
+     * (https://docs.rari.capital/fuse/#exchange-rate)
+     * @param comptrollerAddress address The address of the desired pool's comptroller.
+     * @param tokenAddress address The address of the token to deposit.
+     * @param fTokenAddress address The address of the fToken corresponding to the given token address.
+     * @param amount uin256 The amount of tokens to deposit.
+     */
+    function deposit(
+        address comptrollerAddress,
```

Comment:
Weill these inputs change? 
Whats the strategy for the fuse allocator? any/all treasury assets or fixed ones?

Reply:
If its many assets, do you think its worth tracking which assets we're depositing?

Reply:
Yeah taking a look at it, might be worth to have a mapping from token --> ftoken + comptroller (maybe)

Reply:
Yeah I think that mapping makes sense, the issue is I don't know the fToken addresses yet as Rari's team keeps ghosting me when trying to troubleshoot their contracts. As for the inputs changing, I was told to make it adaptable just in case.

Reply:
I'm not exactly sure the range of assets that we'd want to use. My impression is for the foreseeable future it's just DAI. I could lock it in to be DAI only (we can withdraw and redeploy if we decide to expand) and maybe even default the comptroller to the Olympus Pool address?